.github/workflows/run-e2e-on-fork.yml

```diff
@@ -0,0 +1,144 @@
+name: Run E2E on Fork
+
+concurrency:
+  group: run-e2e-${{ github.event.workflow_run.head_branch }}-${{ github.event.sender.login }}
+  cancel-in-progress: true
+
+on:
+  workflow_run:
+    workflows: [Receive Fork Review]
+    types:
+      - completed
+
+jobs:
+  # check if approver has write/admin permissions
+  check-write-access:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Get user permissions to check if it is either "write" or "admin"'
+        id: get-permissions
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        with:
+          result-encoding: string
+          script: |
+            var { data } = await github.rest.repos.getCollaboratorPermissionLevel({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              username: '${{ github.event.sender.login }}'
+            })
+            console.log('data.permission:', data.permission);
+            return data.permission;
+      - name: 'Fail this workflow if user does not have write permissions'
+        if: ${{ steps.get-permissions.outputs.result != 'write' && steps.get-permissions.outputs.result != 'admin' }}
+        run: |
+          echo "${{ steps.get-permissions.outputs.result }}"
+          exit 1
+  setup:
+    needs: check-write-access
+    runs-on: ubuntu-latest
+    outputs:
+      commit_id: ${{ steps.read-artifact.outputs.result }}
+    # check if receive-fork-review workflow completed successfully
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
Can we move this `if` statement to the `check-write-access` job? So that way, this workflow will simply not run unless the conclusion was success.